File: sdk/src/main/java/org/maibot/sdk/net/HttpRouter.java
package org.maibot.sdk.net;

import io.netty.channel.SimpleChannelInboundHandler;
import io.netty.handler.codec.http.FullHttpRequest;

public abstract class HttpRouter extends SimpleChannelInboundHandler<FullHttpRequest> {
    abstract public void registerProcessor(HttpRequestProcessor processor);
}

File: sdk/src/main/java/org/maibot/sdk/net/HttpRouter.kt
package org.maibot.sdk.net

import io.netty.channel.SimpleChannelInboundHandler
import io.netty.handler.codec.http.FullHttpRequest

abstract class HttpRouter : SimpleChannelInboundHandler<FullHttpRequest>() {
    abstract fun registerProcessor(processor: HttpRequestProcessor)
}
